documentation/api/query/v4/pql.markdown

```diff
@@ -367,16 +383,26 @@ query combined everything before, but with a `resource` subquery for `Package[to
 
 ## Group By
 
-A `group by` clause will condense into a single row all selected rows that share the same values for the grouped expressions.
+As explained above in the `Projections` section, a `group by` clause is
+effected by surrounding a projection in angle brackets. When a function is
+supplied in the projection, however, angle brackets are assumed on the other
+fields. In other words,
+
+    facts[<name>, count(value)] {certname ~ 'web.*'}
+
+and
+
+    facts[name, count(value)] {certname ~ 'web.*'}
 
-For example, to only show a list of fact names, you can group by the `name` field:
+are equivalent, whereas
 
-    facts[name] { group by name }
+    facts[<name>]{}
 
-Combined with aggregate functions, you can effectively rollup of results and aggregate the rolled
-up values, for example to return how many facts exist for each fact name across all facts:
+and
+   
+    facts[name]{}
 
-    facts[name, count(value)] { certname ~ 'web.*' group by name }
+are not, since the first will only return a list of distinct fact names.
```

Comment:
I think we really need to consider making distinct a first class AST operator and connecting <> with that instead of group by. I find it confusing that <> is only independently useful for a fake top-level distinct select, but things like

```
resources[sum(<line>)]
```
are not legal, because it's a group by under the hood. If we had support for distinct we could document it that way, and I would not have an issue with the implicit group by in presence of an aggregate function.